### PR TITLE
Changes for MySQL

### PR DESCRIPTION
**Pull request by ato**
Not ready for merging yet.

* [x] Merge legacy_paths into the blobs table.
* [x] Switch id generation to auto_increment.
* [x] Migrations for MySQL
* [x] Test against mysql
* [x] Tooling to migrate the data (importdb command)
* [x] Document procedure for migrating

### CODE REVIEW

**Comment by nlabuilder** [comment]
SonarQube analysis reported 1 issue

1. MAJOR [Database.java#L123](https://github.com/nla/doss/blob/54d10bab4441dcefbf1d80357ba5d0ecb214b1c7/src/doss/local/Database.java#L123): Method doss.local.Database.version() appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)

**Comment by nlabuilder** [comment]
Build finished.

File: src/doss/local/Database.java [hunk 1]
```diff
@@ -264,10 +264,11 @@ public Long findOrInsertBlobIdByLegacyPath(String legacyPath) {
         return blobId;
     }
 
-    @SqlQuery("SELECT legacy_path FROM legacy_paths WHERE blob_id = :blob_id")
+    @SqlQuery("SELECT legacy_path FROM blobs WHERE blob_id = :blob_id")
     public abstract String locateLegacy(@Bind("blob_id") long blobId);
 
-    @SqlQuery("SELECT blob_id FROM legacy_paths")
+    // FIXME: scalability problems, this should be batched otherwise it'll OOM
```
Comment (brycollins) [inline src/doss/local/Database.java]: Something like this?

@SqlQuery("select count(*) from blobs where legacy_path is not null")
public abstract long getLegacyCount();

@SqlQuery("select blob_id from blobs where lgacy_path is not null limit :limit offset :offset")
public abstract List<Long> listLegacyPathsByBatch(@Bind("limit") long limit, @Bind("offset") long offset);

Then in scrubber get the count, loop by %1000

**Comment by nlabuilder** [comment]
SonarQube analysis reported 6 issues

1. MAJOR [Database.java#L123](https://github.com/nla/doss/blob/b0cc42d03fec92c67aacc3e25e335e455647337a/src/doss/local/Database.java#L123): Method doss.local.Database.version() appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [LocalBlobStore.java#L66](https://github.com/nla/doss/blob/b0cc42d03fec92c67aacc3e25e335e455647337a/src/doss/local/LocalBlobStore.java#L66): Method new doss.local.LocalBlobStore(Path, String) calls InetAddress.getLocalHost(), which may be a security risk [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3AMDM_INETADDRESS_GETLOCALHOST)
1. MAJOR [LocalBlobStore.java#L81](https://github.com/nla/doss/blob/b0cc42d03fec92c67aacc3e25e335e455647337a/src/doss/local/LocalBlobStore.java#L81): Method doss.local.LocalBlobStore.createDefaultConfig(Path) declares throwing two or more exceptions related by inheritance [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_HIERARCHICAL_EXCEPTION_DECLARATION)
1. MAJOR [LocalBlobStore.java#L276](https://github.com/nla/doss/blob/b0cc42d03fec92c67aacc3e25e335e455647337a/src/doss/local/LocalBlobStore.java#L276): doss.local.LocalBlobStore$Tx defines equals and uses Object.hashCode() [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AHE_EQUALS_USE_HASHCODE)
1. MAJOR [LocalBlobStore.java#L326](https://github.com/nla/doss/blob/b0cc42d03fec92c67aacc3e25e335e455647337a/src/doss/local/LocalBlobStore.java#L326): Non derivable method doss.local.LocalBlobStore$Tx$1.commit() declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. MAJOR [LocalBlobStore.java#L405](https://github.com/nla/doss/blob/b0cc42d03fec92c67aacc3e25e335e455647337a/src/doss/local/LocalBlobStore.java#L405): Inconsistent synchronization of doss.core.ManagedTransaction.state; locked 75% of time [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AIS2_INCONSISTENT_SYNC)

**Comment by nlabuilder** [comment]
Build finished.

**Comment by nlabuilder** [comment]
SonarQube analysis reported 8 issues

1. MAJOR [Database.java#L127](https://github.com/nla/doss/blob/b1dcc083d2f5b350bf4d03cd80a5642ceee7b4f1/src/doss/local/Database.java#L127): Method doss.local.Database.version() appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Fsck.java#L61](https://github.com/nla/doss/blob/b1dcc083d2f5b350bf4d03cd80a5642ceee7b4f1/src/doss/local/Fsck.java#L61): Non derivable method doss.local.Fsck$1.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. MAJOR [Fsck.java#L114](https://github.com/nla/doss/blob/b1dcc083d2f5b350bf4d03cd80a5642ceee7b4f1/src/doss/local/Fsck.java#L114): Non derivable method doss.local.Fsck$2.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. MAJOR [LocalBlobStore.java#L66](https://github.com/nla/doss/blob/b1dcc083d2f5b350bf4d03cd80a5642ceee7b4f1/src/doss/local/LocalBlobStore.java#L66): Method new doss.local.LocalBlobStore(Path, String) calls InetAddress.getLocalHost(), which may be a security risk [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3AMDM_INETADDRESS_GETLOCALHOST)
1. MAJOR [LocalBlobStore.java#L81](https://github.com/nla/doss/blob/b1dcc083d2f5b350bf4d03cd80a5642ceee7b4f1/src/doss/local/LocalBlobStore.java#L81): Method doss.local.LocalBlobStore.createDefaultConfig(Path) declares throwing two or more exceptions related by inheritance [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_HIERARCHICAL_EXCEPTION_DECLARATION)
1. MAJOR [LocalBlobStore.java#L275](https://github.com/nla/doss/blob/b1dcc083d2f5b350bf4d03cd80a5642ceee7b4f1/src/doss/local/LocalBlobStore.java#L275): doss.local.LocalBlobStore$Tx defines equals and uses Object.hashCode() [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AHE_EQUALS_USE_HASHCODE)
1. MAJOR [LocalBlobStore.java#L325](https://github.com/nla/doss/blob/b1dcc083d2f5b350bf4d03cd80a5642ceee7b4f1/src/doss/local/LocalBlobStore.java#L325): Non derivable method doss.local.LocalBlobStore$Tx$1.commit() declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. MAJOR [LocalBlobStore.java#L404](https://github.com/nla/doss/blob/b1dcc083d2f5b350bf4d03cd80a5642ceee7b4f1/src/doss/local/LocalBlobStore.java#L404): Inconsistent synchronization of doss.core.ManagedTransaction.state; locked 75% of time [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AIS2_INCONSISTENT_SYNC)

**Comment by nlabuilder** [comment]
Build finished.

**Comment by nlabuilder** [comment]
SonarQube analysis reported 8 issues

1. MAJOR [Database.java#L127](https://github.com/nla/doss/blob/62b02a7ed785b4ed0186a3b4a2c96b56c072249c/src/doss/local/Database.java#L127): Method doss.local.Database.version() appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Fsck.java#L61](https://github.com/nla/doss/blob/62b02a7ed785b4ed0186a3b4a2c96b56c072249c/src/doss/local/Fsck.java#L61): Non derivable method doss.local.Fsck$1.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. MAJOR [Fsck.java#L114](https://github.com/nla/doss/blob/62b02a7ed785b4ed0186a3b4a2c96b56c072249c/src/doss/local/Fsck.java#L114): Non derivable method doss.local.Fsck$2.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. MAJOR [LocalBlobStore.java#L66](https://github.com/nla/doss/blob/62b02a7ed785b4ed0186a3b4a2c96b56c072249c/src/doss/local/LocalBlobStore.java#L66): Method new doss.local.LocalBlobStore(Path, String) calls InetAddress.getLocalHost(), which may be a security risk [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3AMDM_INETADDRESS_GETLOCALHOST)
1. MAJOR [LocalBlobStore.java#L81](https://github.com/nla/doss/blob/62b02a7ed785b4ed0186a3b4a2c96b56c072249c/src/doss/local/LocalBlobStore.java#L81): Method doss.local.LocalBlobStore.createDefaultConfig(Path) declares throwing two or more exceptions related by inheritance [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_HIERARCHICAL_EXCEPTION_DECLARATION)
1. MAJOR [LocalBlobStore.java#L275](https://github.com/nla/doss/blob/62b02a7ed785b4ed0186a3b4a2c96b56c072249c/src/doss/local/LocalBlobStore.java#L275): doss.local.LocalBlobStore$Tx defines equals and uses Object.hashCode() [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AHE_EQUALS_USE_HASHCODE)
1. MAJOR [LocalBlobStore.java#L325](https://github.com/nla/doss/blob/62b02a7ed785b4ed0186a3b4a2c96b56c072249c/src/doss/local/LocalBlobStore.java#L325): Non derivable method doss.local.LocalBlobStore$Tx$1.commit() declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. MAJOR [LocalBlobStore.java#L404](https://github.com/nla/doss/blob/62b02a7ed785b4ed0186a3b4a2c96b56c072249c/src/doss/local/LocalBlobStore.java#L404): Inconsistent synchronization of doss.core.ManagedTransaction.state; locked 75% of time [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AIS2_INCONSISTENT_SYNC)

**Comment by nlabuilder** [comment]
Build finished.

**Comment by nlabuilder** [comment]
SonarQube analysis reported 15 issues
* MAJOR 15 major

#### Top 10 issues

1. MAJOR [Main.java#L178](https://github.com/nla/doss/blob/c2dedd94ff33eb8b456b5e76deed4a4d0229d661/src/doss/Main.java#L178): Method doss.Main$Command$6.execute(Main$Arguments) appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Main.java#L621](https://github.com/nla/doss/blob/c2dedd94ff33eb8b456b5e76deed4a4d0229d661/src/doss/Main.java#L621): Method doss.Main$Command.get(String) throws alternative exception from catch block without history [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ALEST_LOST_EXCEPTION_STACK_TRACE)
1. MAJOR [Main.java#L659](https://github.com/nla/doss/blob/c2dedd94ff33eb8b456b5e76deed4a4d0229d661/src/doss/Main.java#L659): Method doss.Main.readableFileSize(long) creates array using constants [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ASACM_STATIC_ARRAY_CREATED_IN_METHOD)
1. MAJOR [Admin.java#L31](https://github.com/nla/doss/blob/c2dedd94ff33eb8b456b5e76deed4a4d0229d661/src/doss/local/Admin.java#L31): Format string should use %n rather than \n in doss.local.Admin.listContainers() [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AVA_FORMAT_STRING_USES_NEWLINE)
1. MAJOR [Admin.java#L33](https://github.com/nla/doss/blob/c2dedd94ff33eb8b456b5e76deed4a4d0229d661/src/doss/local/Admin.java#L33): Format string should use %n rather than \n in doss.local.Admin.listContainers() [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AVA_FORMAT_STRING_USES_NEWLINE)
1. MAJOR [Admin.java#L106](https://github.com/nla/doss/blob/c2dedd94ff33eb8b456b5e76deed4a4d0229d661/src/doss/local/Admin.java#L106): Method doss.local.Admin.importRemoteDatabase(String) creates array using constants [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ASACM_STATIC_ARRAY_CREATED_IN_METHOD)
1. MAJOR [Admin.java#L133](https://github.com/nla/doss/blob/c2dedd94ff33eb8b456b5e76deed4a4d0229d661/src/doss/local/Admin.java#L133): Method doss.local.Admin.lambda$importRemoteDatabase$0(String[], Handle, Handle, Handle, TransactionStatus) appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Database.java#L127](https://github.com/nla/doss/blob/c2dedd94ff33eb8b456b5e76deed4a4d0229d661/src/doss/local/Database.java#L127): Method doss.local.Database.version() appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Fsck.java#L61](https://github.com/nla/doss/blob/c2dedd94ff33eb8b456b5e76deed4a4d0229d661/src/doss/local/Fsck.java#L61): Non derivable method doss.local.Fsck$1.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. MAJOR [Fsck.java#L114](https://github.com/nla/doss/blob/c2dedd94ff33eb8b456b5e76deed4a4d0229d661/src/doss/local/Fsck.java#L114): Non derivable method doss.local.Fsck$2.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)

**Comment by nlabuilder** [comment]
Build finished.

**Comment by nlabuilder** [comment]
SonarQube analysis reported 15 issues
* MAJOR 15 major

#### Top 10 issues

1. MAJOR [Main.java#L176](https://github.com/nla/doss/blob/06b847bc998482e28c6cbd13041b7c7aacb9af51/src/doss/Main.java#L176): Method doss.Main$Command$6.execute(Main$Arguments) appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Main.java#L633](https://github.com/nla/doss/blob/06b847bc998482e28c6cbd13041b7c7aacb9af51/src/doss/Main.java#L633): Method doss.Main$Command.get(String) throws alternative exception from catch block without history [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ALEST_LOST_EXCEPTION_STACK_TRACE)
1. MAJOR [Main.java#L671](https://github.com/nla/doss/blob/06b847bc998482e28c6cbd13041b7c7aacb9af51/src/doss/Main.java#L671): Method doss.Main.readableFileSize(long) creates array using constants [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ASACM_STATIC_ARRAY_CREATED_IN_METHOD)
1. MAJOR [Admin.java#L32](https://github.com/nla/doss/blob/06b847bc998482e28c6cbd13041b7c7aacb9af51/src/doss/local/Admin.java#L32): Format string should use %n rather than \n in doss.local.Admin.listContainers() [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AVA_FORMAT_STRING_USES_NEWLINE)
1. MAJOR [Admin.java#L34](https://github.com/nla/doss/blob/06b847bc998482e28c6cbd13041b7c7aacb9af51/src/doss/local/Admin.java#L34): Format string should use %n rather than \n in doss.local.Admin.listContainers() [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AVA_FORMAT_STRING_USES_NEWLINE)
1. MAJOR [Admin.java#L107](https://github.com/nla/doss/blob/06b847bc998482e28c6cbd13041b7c7aacb9af51/src/doss/local/Admin.java#L107): Method doss.local.Admin.importRemoteDatabase(String) creates array using constants [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ASACM_STATIC_ARRAY_CREATED_IN_METHOD)
1. MAJOR [Admin.java#L134](https://github.com/nla/doss/blob/06b847bc998482e28c6cbd13041b7c7aacb9af51/src/doss/local/Admin.java#L134): Method doss.local.Admin.lambda$importRemoteDatabase$0(String[], Handle, Handle, Handle, TransactionStatus) appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Database.java#L127](https://github.com/nla/doss/blob/06b847bc998482e28c6cbd13041b7c7aacb9af51/src/doss/local/Database.java#L127): Method doss.local.Database.version() appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Fsck.java#L61](https://github.com/nla/doss/blob/06b847bc998482e28c6cbd13041b7c7aacb9af51/src/doss/local/Fsck.java#L61): Non derivable method doss.local.Fsck$1.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. MAJOR [Fsck.java#L114](https://github.com/nla/doss/blob/06b847bc998482e28c6cbd13041b7c7aacb9af51/src/doss/local/Fsck.java#L114): Non derivable method doss.local.Fsck$2.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)

**Comment by nlabuilder** [comment]
Build finished.

**Comment by nlabuilder** [comment]
SonarQube analysis reported 15 issues
* MAJOR 15 major

#### Top 10 issues

1. MAJOR [Main.java#L176](https://github.com/nla/doss/blob/afa531743a217de294c19ec1189c6be03e89a5ca/src/doss/Main.java#L176): Method doss.Main$Command$6.execute(Main$Arguments) appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Main.java#L633](https://github.com/nla/doss/blob/afa531743a217de294c19ec1189c6be03e89a5ca/src/doss/Main.java#L633): Method doss.Main$Command.get(String) throws alternative exception from catch block without history [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ALEST_LOST_EXCEPTION_STACK_TRACE)
1. MAJOR [Main.java#L671](https://github.com/nla/doss/blob/afa531743a217de294c19ec1189c6be03e89a5ca/src/doss/Main.java#L671): Method doss.Main.readableFileSize(long) creates array using constants [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ASACM_STATIC_ARRAY_CREATED_IN_METHOD)
1. MAJOR [Admin.java#L32](https://github.com/nla/doss/blob/afa531743a217de294c19ec1189c6be03e89a5ca/src/doss/local/Admin.java#L32): Format string should use %n rather than \n in doss.local.Admin.listContainers() [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AVA_FORMAT_STRING_USES_NEWLINE)
1. MAJOR [Admin.java#L34](https://github.com/nla/doss/blob/afa531743a217de294c19ec1189c6be03e89a5ca/src/doss/local/Admin.java#L34): Format string should use %n rather than \n in doss.local.Admin.listContainers() [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AVA_FORMAT_STRING_USES_NEWLINE)
1. MAJOR [Admin.java#L107](https://github.com/nla/doss/blob/afa531743a217de294c19ec1189c6be03e89a5ca/src/doss/local/Admin.java#L107): Method doss.local.Admin.importRemoteDatabase(String) creates array using constants [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ASACM_STATIC_ARRAY_CREATED_IN_METHOD)
1. MAJOR [Admin.java#L141](https://github.com/nla/doss/blob/afa531743a217de294c19ec1189c6be03e89a5ca/src/doss/local/Admin.java#L141): Method doss.local.Admin.lambda$importRemoteDatabase$0(String[], Handle, Handle, Handle, TransactionStatus) appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Database.java#L127](https://github.com/nla/doss/blob/afa531743a217de294c19ec1189c6be03e89a5ca/src/doss/local/Database.java#L127): Method doss.local.Database.version() appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Fsck.java#L61](https://github.com/nla/doss/blob/afa531743a217de294c19ec1189c6be03e89a5ca/src/doss/local/Fsck.java#L61): Non derivable method doss.local.Fsck$1.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. MAJOR [Fsck.java#L114](https://github.com/nla/doss/blob/afa531743a217de294c19ec1189c6be03e89a5ca/src/doss/local/Fsck.java#L114): Non derivable method doss.local.Fsck$2.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)

**Comment by nlabuilder** [comment]
Build finished.

**Comment by nlabuilder** [comment]
SonarQube analysis reported 57 issues
* MAJOR 57 major

#### Top 10 issues

1. MAJOR [Main.java#L176](https://github.com/nla/doss/blob/ace3269f8142d0395f28e9c1a2cff0e007c6d7e5/src/doss/Main.java#L176): Method doss.Main$Command$6.execute(Main$Arguments) appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Main.java#L633](https://github.com/nla/doss/blob/ace3269f8142d0395f28e9c1a2cff0e007c6d7e5/src/doss/Main.java#L633): Method doss.Main$Command.get(String) throws alternative exception from catch block without history [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ALEST_LOST_EXCEPTION_STACK_TRACE)
1. MAJOR [Main.java#L671](https://github.com/nla/doss/blob/ace3269f8142d0395f28e9c1a2cff0e007c6d7e5/src/doss/Main.java#L671): Method doss.Main.readableFileSize(long) creates array using constants [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ASACM_STATIC_ARRAY_CREATED_IN_METHOD)
1. MAJOR [Admin.java#L32](https://github.com/nla/doss/blob/ace3269f8142d0395f28e9c1a2cff0e007c6d7e5/src/doss/local/Admin.java#L32): Format string should use %n rather than \n in doss.local.Admin.listContainers() [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AVA_FORMAT_STRING_USES_NEWLINE)
1. MAJOR [Admin.java#L34](https://github.com/nla/doss/blob/ace3269f8142d0395f28e9c1a2cff0e007c6d7e5/src/doss/local/Admin.java#L34): Format string should use %n rather than \n in doss.local.Admin.listContainers() [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AVA_FORMAT_STRING_USES_NEWLINE)
1. MAJOR [Admin.java#L107](https://github.com/nla/doss/blob/ace3269f8142d0395f28e9c1a2cff0e007c6d7e5/src/doss/local/Admin.java#L107): Method doss.local.Admin.importRemoteDatabase(String) creates array using constants [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ASACM_STATIC_ARRAY_CREATED_IN_METHOD)
1. MAJOR [Admin.java#L141](https://github.com/nla/doss/blob/ace3269f8142d0395f28e9c1a2cff0e007c6d7e5/src/doss/local/Admin.java#L141): Method doss.local.Admin.lambda$importRemoteDatabase$0(String[], Handle, Handle, Handle, TransactionStatus) appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Database.java#L127](https://github.com/nla/doss/blob/ace3269f8142d0395f28e9c1a2cff0e007c6d7e5/src/doss/local/Database.java#L127): Method doss.local.Database.version() appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Fsck.java#L61](https://github.com/nla/doss/blob/ace3269f8142d0395f28e9c1a2cff0e007c6d7e5/src/doss/local/Fsck.java#L61): Non derivable method doss.local.Fsck$1.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. MAJOR [Fsck.java#L114](https://github.com/nla/doss/blob/ace3269f8142d0395f28e9c1a2cff0e007c6d7e5/src/doss/local/Fsck.java#L114): Non derivable method doss.local.Fsck$2.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)

**Comment by nlabuilder** [comment]
Build finished.

**Comment by nlabuilder** [comment]
SonarQube analysis reported 57 issues
* MAJOR 57 major

#### Top 10 issues

1. MAJOR [Main.java#L176](https://github.com/nla/doss/blob/aa37803603bd7780fbc27020e32971800241a2e9/src/doss/Main.java#L176): Method doss.Main$Command$6.execute(Main$Arguments) appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Main.java#L633](https://github.com/nla/doss/blob/aa37803603bd7780fbc27020e32971800241a2e9/src/doss/Main.java#L633): Method doss.Main$Command.get(String) throws alternative exception from catch block without history [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ALEST_LOST_EXCEPTION_STACK_TRACE)
1. MAJOR [Main.java#L671](https://github.com/nla/doss/blob/aa37803603bd7780fbc27020e32971800241a2e9/src/doss/Main.java#L671): Method doss.Main.readableFileSize(long) creates array using constants [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ASACM_STATIC_ARRAY_CREATED_IN_METHOD)
1. MAJOR [Admin.java#L32](https://github.com/nla/doss/blob/aa37803603bd7780fbc27020e32971800241a2e9/src/doss/local/Admin.java#L32): Format string should use %n rather than \n in doss.local.Admin.listContainers() [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AVA_FORMAT_STRING_USES_NEWLINE)
1. MAJOR [Admin.java#L34](https://github.com/nla/doss/blob/aa37803603bd7780fbc27020e32971800241a2e9/src/doss/local/Admin.java#L34): Format string should use %n rather than \n in doss.local.Admin.listContainers() [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=findbugs%3AVA_FORMAT_STRING_USES_NEWLINE)
1. MAJOR [Admin.java#L107](https://github.com/nla/doss/blob/aa37803603bd7780fbc27020e32971800241a2e9/src/doss/local/Admin.java#L107): Method doss.local.Admin.importRemoteDatabase(String) creates array using constants [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ASACM_STATIC_ARRAY_CREATED_IN_METHOD)
1. MAJOR [Admin.java#L141](https://github.com/nla/doss/blob/aa37803603bd7780fbc27020e32971800241a2e9/src/doss/local/Admin.java#L141): Method doss.local.Admin.lambda$importRemoteDatabase$0(String[], Handle, Handle, Handle, TransactionStatus) appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Database.java#L127](https://github.com/nla/doss/blob/aa37803603bd7780fbc27020e32971800241a2e9/src/doss/local/Database.java#L127): Method doss.local.Database.version() appears to call the same method on the same object redundantly [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Fsck.java#L61](https://github.com/nla/doss/blob/aa37803603bd7780fbc27020e32971800241a2e9/src/doss/local/Fsck.java#L61): Non derivable method doss.local.Fsck$1.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)
1. MAJOR [Fsck.java#L114](https://github.com/nla/doss/blob/aa37803603bd7780fbc27020e32971800241a2e9/src/doss/local/Fsck.java#L114): Non derivable method doss.local.Fsck$2.visitFile(Path, BasicFileAttributes) declares throwing an exception that isn't thrown [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=fb-contrib%3ABED_BOGUS_EXCEPTION_DECLARATION)

**Comment by nlabuilder** [comment]
Build finished.

**Pull request merged**